src/librustc_mir/build/scope.rs

```diff
@@ -428,6 +428,21 @@ impl<'a, 'gcx, 'tcx> Builder<'a, 'gcx, 'tcx> {
         self.scopes.last().expect("topmost_scope: no scopes present").extent
     }
 
+    /// Returns the scope that we should use as the lifetime of an
+    /// operand. Basically, an operand must live until it is consumed.
+    /// This is similar to, but not quite the same as, the temporary
+    /// scope (which can be larger or smaller).
```

Comment:
More concretely, if we have `foo(Box::new(2), panic!())`, because we evaluate arguments from left to right, we allocate the `Box` and then free it (because `foo` can also panic, we actually emit a drop flag there - but a "sufficiently smart" drop unswitching pass could eliminate that). We also emit a `drop` after the call on the "normal" path, but that drop is a no-op because the value is already consumed.